src/Akka.Persistence.Hosting.Tests/InMemoryPersistenceSpecs.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.Event;
+using Akka.Hosting;
+using Akka.TestKit.Xunit2.Internals;
+using FluentAssertions;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Akka.Persistence.Hosting.Tests
+{
+    public class InMemoryPersistenceSpecs
+    {
+        
+        private readonly ITestOutputHelper _output;
+
+        public InMemoryPersistenceSpecs(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        public sealed class MyPersistenceActor : ReceivePersistentActor
+        {
+            private List<int> _values = new List<int>();
+
+            public MyPersistenceActor(string persistenceId)
+            {
+                PersistenceId = persistenceId;
+                
+                Recover<SnapshotOffer>(offer =>
+                {
+                    if (offer.Snapshot is IEnumerable<int> ints)
+                    {
+                        _values = new List<int>(ints);
+                    }
+                });
+                
+                Recover<int>(i =>
+                {
+                    _values.Add(i);
+                });
+                
+                Command<int>(i =>
+                {
+                    Persist(i, i1 =>
+                    {
+                        _values.Add(i);
+                        if (LastSequenceNr % 2 == 0)
+                        {
+                            SaveSnapshot(_values);
+                        }
+                        Sender.Tell("ACK");
+                    });
+                });
+
+                Command<string>(str => str.Equals("getall"), s =>
+                {
+                    Sender.Tell(_values.ToArray());
+                });
+                
+                Command<SaveSnapshotSuccess>(s => {});
+            }
+
+            public override string PersistenceId { get; }
+        }
+        
+        public static async Task<IHost> StartHost(Action<IServiceCollection> testSetup)
+        {
+            var host = new HostBuilder()
+                .ConfigureServices(testSetup).Build();
+        
+            await host.StartAsync();
+            return host;
+        }
+
+        [Fact]
+        public async Task Should_Start_ActorSystem_wth_InMemory_Persistence()
```

Comment:
End2End spec to verify we have the right plugins running, including some actor start / stop specs.